jerry-core/ecma/operations/ecma-regexp-object.cpp

```diff
@@ -119,6 +119,101 @@ re_parse_regexp_flags (ecma_string_t *flags_str_p, /**< Input string with flags
   return ret_value;
 } /* re_parse_regexp_flags  */
 
+/*
+ * Initialize properties of RegExp instance.
+ */
+void
+re_initialize_props (ecma_object_t *re_obj_p,
+                     ecma_string_t *source_p,
+                     uint8_t flags)
+{
+ /* Set source property. ECMA-262 v5, 15.10.7.1 */
+  ecma_string_t *magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_SOURCE);
+  ecma_property_t *prop_p = ecma_find_named_property (re_obj_p, magic_string_p);
+
+  if (!prop_p)
+  {
+    prop_p = ecma_create_named_data_property (re_obj_p,
+                                              magic_string_p,
+                                              false, false, false);
+  }
+
+  ecma_deref_ecma_string (magic_string_p);
+  JERRY_ASSERT (prop_p->type == ECMA_PROPERTY_NAMEDDATA);
+  ecma_named_data_property_assign_value (re_obj_p,
+                                         prop_p,
+                                         ecma_make_string_value (source_p));
+
+  ecma_simple_value_t prop_value;
+
+  /* Set global property. ECMA-262 v5, 15.10.7.2*/
+  magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_GLOBAL);
+  prop_p = ecma_find_named_property (re_obj_p, magic_string_p);
+
+  if (!prop_p)
+  {
+    prop_p = ecma_create_named_data_property (re_obj_p,
+                                              magic_string_p,
+                                              false, false, false);
+  }
+
+  ecma_deref_ecma_string (magic_string_p);
+  prop_value = flags & RE_FLAG_GLOBAL ? ECMA_SIMPLE_VALUE_TRUE : ECMA_SIMPLE_VALUE_FALSE;
+  JERRY_ASSERT (prop_p->type == ECMA_PROPERTY_NAMEDDATA);
+  ecma_set_named_data_property_value (prop_p, ecma_make_simple_value (prop_value));
+
+  /* Set ignoreCase property. ECMA-262 v5, 15.10.7.3*/
+  magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_IGNORECASE_UL);
+  prop_p = ecma_find_named_property (re_obj_p, magic_string_p);
+
+  if (!prop_p)
```

Comment:
The properties already exist when we call `compile` on a RegExp object. In that case we only have to re-initialize their values.